### PR TITLE
feat: parallelize moon test in pkg level

### PR DESCRIPTION
**Pull request by Young-Flash**
test at [core](https://github.com/moonbitlang/core) with the following result

before: 2.2 seconds on average (after warmup)
image

after: 0.2 seconds on average (after warmup)
image

part of https://github.com/moonbitlang/moon/issues/47

### CODE REVIEW

**Comment by Young-Flash** [comment]
test failure supported to be fixed in the next release

File: crates/moon/src/cli/test.rs [hunk 1]
```diff
@@ -78,6 +78,9 @@ pub struct TestSubcommand {
 
     #[clap(flatten)]
     pub auto_sync_flags: AutoSyncFlags,
+
+    #[clap(long)]
+    pub no_parallelize: bool,
```
Comment (lijunchen) [inline crates/moon/src/cli/test.rs:83]: just use `no-parallel`

Maybe we need `--jobs` flag?

File: crates/moonbuild/src/entry.rs [hunk 2]
```diff
@@ -321,101 +328,132 @@ pub fn run_test(
...
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()?;
+
+    let mut handlers = vec![];
 
     for runnable_artifact in runnable_artifacts.iter() {
         let p = Path::new(runnable_artifact);
 
+        let passed = passed.clone();
```
Comment (lijunchen) [inline crates/moonbuild/src/entry.rs:346]: use `Arc::clone` instead of `.clone`?

**Review by lijunchen — APPROVED** [review]

File: crates/moonbuild/src/entry.rs (hunk 2 above)
Comment (Young-Flash) [reply crates/moonbuild/src/entry.rs:346]: seems no difference here, ide inlay hints will tell us it's `Arc` wrapped

File: crates/moon/src/cli/test.rs (hunk 1 above)
Comment (Young-Flash) [reply crates/moon/src/cli/test.rs:83]: will make at next PR

**Pull request merged**